draft-ietf-tls-tls13.md

```diff
@@ -2323,102 +2315,138 @@ The "extension_data" field of this extension contains a
 
 %%% Signature Algorithm Extension
        enum {
-           none(0),
-           md5_RESERVED(1),
-           sha1(2),
-           sha224_RESERVED(3),
-           sha256(4), sha384(5), sha512(6),
-           (255)
-       } HashAlgorithm;
+           // RSASSA-PKCS-v1_5 algorithms.
+           rsa_pkcs1_sha1 (0x0101),
+           rsa_pkcs1_sha256 (0x0301),
+           rsa_pkcs1_sha384 (0x0401),
+           rsa_pkcs1_sha512 (0x0601),
+
+           // DSA algorithms (deprecated).
+           dsa_sha1 (0x0102),
+           dsa_sha256 (0x0302),
+           dsa_sha384 (0x0402),
+           dsa_sha512 (0x0602),
+
+           // ECDSA algorithms.
+           ecdsa_secp256r1_sha256 (0x0303),
+           ecdsa_secp384r1_sha384 (0x0403),
+           ecdsa_secp521r1_sha512 (0x0603),
+
+           // RSASSA-PSS algorithms.
+           rsa_pss_sha256 (0x0700),
+           rsa_pss_sha384 (0x0701),
+           rsa_pss_sha512 (0x0702),
+
+           // EdDSA algorithms.
+           ed25519 (0x0703),
+           ed448 (0x0704),
 
-       enum {
-           anonymous_RESERVED(0),
-           rsa(1),
-           dsa(2),
-           ecdsa(3),
-           rsapss(4),
-           eddsa(5),
-           (255)
+           // Reserved Code Points.
+           obsolete_RESERVED (0x0000..0x0100),
+           obsolete_RESERVED (0x0103..0x0300),
+           obsolete_RESERVED (0x0304..0x0400),
+           obsolete_RESERVED (0x0404..0x0600),
+           obsolete_RESERVED (0x0604..0x06FF),
+           private_use (0xFE00..0xFFFF),
+           (0xFFFF)
        } SignatureAlgorithm;
```

Comment:
I don't think we should be doing out-of-order enums and I don't want the obsolete ranges to be unnamed either. I would much rather we stick to the same ordering, as I first proposed.
https://www.ietf.org/mail-archive/web/tls/current/msg18035.html

If we really want a list sorted by cert, we can stick one in an appendix.